generator/src/main/java/com/scottlogic/deg/generator/generation/databags/StandardRowSpecDataBagSourceFactory.java

```diff
@@ -6,44 +6,34 @@
 import com.scottlogic.deg.generator.fieldspecs.RowSpec;
 import com.scottlogic.deg.generator.generation.FieldSpecValueGenerator;
 import com.scottlogic.deg.generator.generation.GenerationConfig;
+import com.scottlogic.deg.generator.generation.combinationstrategies.CombinationStrategy;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Stream;
 
 public class StandardRowSpecDataBagSourceFactory implements RowSpecDataBagSourceFactory {
     private final FieldSpecValueGenerator generator;
+    private final CombinationStrategy combinationStrategy;
 
     @Inject
-    public StandardRowSpecDataBagSourceFactory (
-        FieldSpecValueGenerator generator) {
+    public StandardRowSpecDataBagSourceFactory(
+        FieldSpecValueGenerator generator,
+        CombinationStrategy combinationStrategy) {
         this.generator = generator;
+        this.combinationStrategy = combinationStrategy;
     }
 
-    public DataBagSource createDataBagSource(RowSpec rowSpec){
-
-        List<DataBagSource> fieldDataBagSources = new ArrayList<>(rowSpec.getFields().size());
+    public Stream<DataBag> createDataBags(RowSpec rowSpec){
+        List<Stream<DataBag>> fieldDataBagSources = new ArrayList<>(rowSpec.getFields().size());
```

Comment:
This implementation could look/read a bit better if you used streams and mapped things as you go.

Reply:
i did this originally, but didn't like how it looked. i'll give it another go